cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -139,14 +139,14 @@ LedgerLoop:
 	for ledgerSeq := start.LedgerSequence; ledgerSeq <= int32(ledgerRange.LastLedger.Sequence); ledgerSeq++ {
 		// Get ledger close meta from db
 		ledger, found, err := h.ledgerReader.GetLedger(ctx, uint32(ledgerSeq))
-		if (err != nil) || (!found) {
-			code := jrpc2.InternalError
-			if err == nil {
-				err = errors.Errorf("ledger close meta not found: %d", ledgerSeq)
-				code = jrpc2.InvalidParams
+		if !found {
```

Comment:
you need to check `err` first because it is possible that `err` is not nil `found` is false

Reply:
Ah the default case. Gotcha